### ISSUE TITLE
Create connection to weather API from Java

### ISSUE THREAD

**Issue by frksteenhoff** [frksteenhoff/ClimApp#1]
I am having trouble finding the relevant data to fetch and where to get it from. The entire website architecture is really old and it does not seem to have any API to access? Possibly try something else? Did Hein have a different weather API we could test?

**Comment by Chuansi-LU** [comment]
Hi Henriette, Great work! I have just forwarded an email to your gmail address with regard to the help you wanted to get access to weather data.

**Comment by Chuansi-LU** [comment]
Hi Henriette, Hein and Boris (Warm sweater app) in The Netherlands used the following weather data:
www.openweathermap.org 
MeteoSwiss and Lars recommend German DWD weather data (the data format will be changed in 2018).
https://www.dwd.de/EN/ourservices/opendata/opendata.html
In the published paper, the Global Forecast System GFS-NOAA weather data are used.
Perhaps each of these weather data has pros and cons.

**Comment by frksteenhoff** [comment]
Having looked into the two options, I must say that using Open Weather Data is vastly more convenient than DWD. I will use data from Open Weather Data in the prototype as their API is much easier to work with and way better technologically with a higher number of options for getting the needed information.

**Comment by frksteenhoff** [comment]
This service can however only fetch data every 10 minutes from one API key. This is something we have to keep in mind.

**Comment by Chuansi-LU** [comment]
Very good you have compared these two weather data services. I have also read some of the features of the Open Weather Data. It provides 5 days forecast with 3 hour interval (this seems good enough for our ClimApp). Is this interval related to "only fetch data every 10 minutes"?

**Comment by Chuansi-LU** [comment]
Is the mean radiant temperature (solar radiation) and air humidity (either relative humidity, partial water vapour pressure, or dew Point) available from the Open Weather Data?

**Comment by frksteenhoff** [comment]
Hi Chuansi, as seen in following link, you get different information as shown in the data structure. As it is fetched directly from the stations, the data is more frequent. As I understand it, only the forecasts are on a 3 hour basis. http://openweathermap.org/current

**Comment by Nybo72** [comment]
I suggest we ask MeteoSwiss how they extract data (as they have developed state of the art for that part) - also that we decide the time resolution and time before alert (e.g. limit to < 24 h or < 48 h) people wont remember/use for more than one day at the time anyway -> making/taking these decisions may simplify some of the processes a lot.

**Comment by Nybo72** [comment]
DMI also promissed to help with the data extraction and they use thede open sources for identifying cyties http://www.geonames.org/ and although they use a different model for 0-48 h they rely on the ECMWF model for most of Europe (same model/data set as the DWD and YR) hence they must know how we combine info to extract the 4 weather parameters - but suggest we ask MeteoSwiss first

**Comment by frksteenhoff** [comment]
From YR you can extract this hourly weather forecast as an XML file, http://www.yr.no/place/Norway/Telemark/Sauherad/Gvarv/forecast_hour_by_hour.xml

You can also get this information as a 6 hour forecast and for the upcoming 90 minutes (which I think could be a smart solution for the daily updates)

However, compared to YR, I actually think OpeanWeatherMap has a better solution as their forecast also includes humidity, pressure etc. in one XML file that is smaller and therefore takes less time to calculate. 

`{"coord":{"lon":139,"lat":35},
"sys":{"country":"JP","sunrise":1369769524,"sunset":1369821049},
"weather":[{"id":804,"main":"clouds","description":"overcast clouds","icon":"04n"}],
"main":{"temp":289.5,"humidity":89,"pressure":1013,"temp_min":287.04,"temp_max":292.04},
"wind":{"speed":7.31,"deg":187.002},
"rain":{"3h":0},
"clouds":{"all":92},
"dt":1369824698,
"id":1851632,
"name":"Shuzenji",
"cod":200}`

TL;DR:
For the prototype, I will use this information when trying to implement it in the app, as the connections to most APIs consist of giving an API key and a URL and your location as lon/lat pair. I think it will be possible to make a generic solution/class to handle this and then make slight modifications to fit/adapt to whatever other API we might find better later on.

**Comment by Nybo72** [comment]
Har du 2 min på tlf så kan jeg hurtigt hjælpe

**Comment by frksteenhoff** [comment]
Hej Lars, sorry. Jeg har lige været lidt i min egen kode-verden. Hjælp ifm. hvilket noget af det? Du giver bar et kald hvis det er (+45 26661248)

**Comment by Chuansi-LU** [comment]
Hi Lars and Henriette, 
1. As the first step is to make a prototype and a trial Connection to weather data, I think it is worth of trying Open Weather Data. During the course of the develpment and improvement, it can be switched and modified to different weather APIs if we find better ones.
2. At the same time, Lars: could you please Contact MeteoSwish about the weather API.
3. Some information about ECMWF model: The Integrated Forecast System (IFS) is an operational global meteorological forecasting model. IFS is developed and maintained by the European Centre for Medium-Range Weather Forecasts (ECMWF) based in Reading, England. Information from the IFS is proprietary and copyrighted, though the ECMWF has made a limited amount of the model's most important calculations available to the public; this public data is, by declaration of the ECMWF, in the public domain. 
4.  ECMWF focuses on medium-range weather forecasts, which is needed for Heat-Shield Project, but not for ClimApp Project. ClimApp focuses on short-term forefasts, short-term forecast data are already openly available, that is also what I heard from Christoph in Durham. MeteoSwiss has been working on Medium range forecast for Heat-shield Project. They don't do much about short term forecast.

**Comment by Chuansi-LU** [comment]
Open Weather Data are collected and then provided for open access by an IT Company. This is probably why API is easier to use. The weather data include ECMWF data, NOAA data, Canadian Meteorological Centre data, Japan meteorological data, and other weather sources.

**Comment by frksteenhoff** [comment]
Thank you Chuansi. I am almost done with implementing getting forecast data from Open Weather Data. I am still working on getting the location directly from the phone, but using "hard coded" values for testing is working.

**Comment by Nybo72** [comment]
It is fine to develop the prototype with this (and it may also be fine for the permanent solution) – However, I suggest (as agreed with Jørn and Chuansi on meeting Friday) that we (Henriette and I) talk to meteoSwiss to ensure that this forecast is validated for all the 4 parameters of interest (although we are using short term forecasts they have done this job for the ECMWF model (and if all the European station in open rely on that then it is fine -> but we should coordinate with them + use data endorsed by the ERA4CS so we don’t compromise “met-standards”.

I will give you a call tomorrow Henriette – OK around 12 ?

Bw Lars

**Comment by Chuansi-LU** [comment]
Henriette, great work during weekend!
Yes, Lars, we need to ask ERA4CS for advice too. I will participate a meeting organized by ERA4CS for all project coordinators on 30th of November in Brussels.

Best regards,
Chuansi
